packages/web/src/shared/components/form/ButtonSelectorInput.tsx

```diff
@@ -3,33 +3,46 @@ import { useFormContext } from 'react-hook-form';
 
 interface ButtonSelectorContextValue {
   name: string;
+  value?: string;
+  onChange?(value: string): void;
   defaultValue?: string;
-  required?: boolean;
+  hasErrors: boolean;
 }
 
 const ButtonSelectorContext = createContext<ButtonSelectorContextValue | null>(
   null
 );
 
 export interface ButtonSelectorInputProps
-  extends Omit<ComponentProps<'fieldset'>, 'defaultValue'> {
+  extends Omit<
+    ComponentProps<'fieldset'>,
+    'defaultValue' | 'value' | 'onChange'
+  > {
   name: string;
+  value?: string;
+  onChange?(value: string): void;
   defaultValue?: string;
-  required?: boolean;
 }
 
 export function ButtonSelectorInput({
+  value,
+  onChange,
   children,
   name,
   defaultValue,
-  required,
   ...props
 }: ButtonSelectorInputProps) {
   const formContext = useFormContext();
-  const hasErrors = !!formContext?.formState.errors[name];
+  const hasErrors = !!(
+    formContext &&
+    name &&
+    formContext.getFieldState(name).error
+  );
```

Comment:
Inside an input component, we can still use the form context to get the error state in order to render the red border because this is filtered by whether the name of the input (if there is one) is in the form or not.